tests/testsuite/patch.rs

```diff
@@ -3028,3 +3028,49 @@ foo v0.0.0 ([ROOT]/foo)
 
     assert_eq!(p.read_file("Cargo.lock"), p.read_file("Cargo.lock.orig"));
 }
+
+#[cargo_test]
+fn patch_with_base() {
```

Comment:
Do we also need a test for `[patch]` in `.cargo/config.toml`?

Reply:
Good catch - I'd tried adding this and implementing it, but it raised a few big issues:
1. Configuration doesn't have a `cargo-features` key (as I understand it), so how do we handle unstable features? (Just allow using base paths? Require that every manifest using this configuration allow base paths?)
2. How to handle the `workspace` built-in? (block it?)
3. `patch` and `replace` have a unified code path for resolving dependencies, so do I need to differentiate to only permit `base` in `patch` tables?
4. Do I need to add back the code to handle base paths in `to_dependency_source_id`?

Reply:
I might need to move support for `[patch]` in config files into a new PR. Or maybe drop support for `[patch]` given the complications it causes...

Reply:
> I might need to move support for [patch] in config files into a new PR. Or maybe drop support for [patch] given the complications it causes...

I think its doable but I also agree we should probably split it out at this point so it doesn't block the rest of your work

> Configuration doesn't have a cargo-features key (as I understand it), so how do we handle unstable features? (Just allow using base paths? Require that every manifest using this configuration allow base paths?)

This would use config's unstable mechanism,

> How to handle the workspace built-in? (block it?)

config is relative to itself and is independent of workspaces, so it should fail

> patch and replace have a unified code path for resolving dependencies, so do I need to differentiate to only permit base in patch tables?

Unsure.

> Do I need to add back the code to handle base paths in to_dependency_source_id?

We need to handle patch somewhere.  Where I don't care but preferably we don't change the normalization from how it is now in this PR.

Reply:
I've added "`[patch]` table in config support" as one of the items to implement in the tracking issue

Reply:
ok